webapp/security/views.py

```diff
@@ -536,3 +513,57 @@ def delete_cve(cve_id):
         return flask.jsonify({"message": error.orig.args[0]}), 400
 
     return flask.jsonify({"message": "CVE deleted succesfully"}), 200
+
+
+@authorization_required
+def bulk_upsert_cve():
+    """
+    Receives a PUT request from load_cve.py
+    Parses the object and bulk inserts or updates
+    @returns 3 lists of CVEs, created CVEs, updated CVEs and failed CVEs
+    """
+
+    cves_data = flask.request.json
+    create = []
+    update = []
+
+    if not type(cves_data) == list:
+        return (
+            flask.jsonify(
+                {"message": "Please, submit a list (array) of CVEs"}
+            ),
+            400,
+        )
+
+    for data in cves_data:
+        cve = db_session.query(CVE).get(data["id"])
+        if cve:
+            update.append(data)
+        else:
+            create.append(data)
+
+    updated = update_cve(update)
+    created = create_cve(create)
```

Comment:
Out of curiosity, what happens if you run `create_cve` for a CVE that already exists? Is there any chance that sqlalchemy simply takes care of it?

Reply:
Do you mean `add_all` on existent CVEs? My code will never run `create_cve` on existent CVEs, because it's been filtered by line 539, this is done on purpose, because SQLAlchemy's `add_all` needs the original object to update. If you try to create the way I did it in the `create_cve`. it will complain with `Duplicate key error` so you have to feed it with an existent object, with all the fields correctly mapped. Which is why I separated it in 2 `add_all`.